app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -265,12 +276,47 @@ function isMatrix(data: object): data is LegacyMatrix {
   return json.every((d) => d.length === firstLen)
 }
 
-function toField(name: string): ColDef {
-  return { field: name }
+function toField(name: string, valType?: string | undefined | null, displayValue?: any): ColDef {
```

Comment:
If you have added `ValueType` type above, could the `toField` function take `valueType?: ValueType | null` instead? It feels like it could be cleaner that way.

Reply:
yeah great spot!!